src/common/soc.c

```diff
@@ -78,6 +78,16 @@ void fill_soc(struct system_on_chip* soc, char* soc_name, SOC soc_model, int32_t
   }
 }
 
+VENDOR try_match_soc_vendor_name(char* vendor_name)
+{
+  for(size_t i=0; i < sizeof(soc_trademark_string)/sizeof(soc_trademark_string[0]); i++) {
+    if(soc_trademark_string[i] && strstr(vendor_name, soc_trademark_string[i]) != NULL) {
```

Comment:
Why do we need the first condition?

Reply:
This was due to the fact that `soc_trademark_string` is not a complete and contiguous array and I was getting segfaults otherwise.
https://github.com/Dr-Noob/cpufetch/blob/977c35a9af4b82ddff27cd471ad7e98dae9e7ef9/src/common/soc.c#L12-L33

Looking at it now, it's just that there is no entry for `SOC_VENDOR_UNKNOWN` so `soc_trademark_string[0]` is a null string.

I could also just iterate starting at 1 to skip that entry(`for(size_t i=1;...`)